bundles/admin/admin-layereditor/view/Flyout.js

```diff
@@ -16,6 +16,8 @@ export class LayerEditorFlyout extends ExtraFlyout {
         this.dataProviders = [];
         this.mapLayerGroups = [];
         this.service = new AdminLayerFormService(() => this.update());
+        this.mapLayerService = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
+        this.mapLayerService.addStateListener(() => this.update());
```

Comment:
Use as observable. `this.mapLayerService.on('some-event-name', () => this.update())`

Reply:
I tried to follow similar notification mechanism that is used with AdminLayerFormService in same flyout and for this reason implementation was done as it was. Implementation has now been changed as suggested in review comments.